docs/eloquent-models/schema-builder.txt

```diff
@@ -0,0 +1,388 @@
+
+Create a Migration Class
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can create migration classes manually or generate them by using the
+``php artisan make:migration`` command. If you generate them, you must make the
+following changes to perform the schema changes on your MongoDB database:
+
+- Replace the ``Illuminate\Database\Schema\Blueprint`` import with
+  ``MongoDB\Laravel\Schema\Blueprint`` if it is referenced in your migration
+- Specify ``mongodb`` in the ``$connection`` field
+- Use only commands and syntax supported by {+odm-short+}
+
+.. tip::
+
+   Make sure you set ``DB_CONNECTION=mongodb`` in your ``.env`` configuration
+   file so that PHP artisan performs the migration on the correct database.
+
+The following example migration creates a collection and index when you run
+the migration and drops them when you roll back the migration:
+
+.. literalinclude:: /includes/schema-builder/astronauts_migration.php
+   :dedent:
+   :language: php
+   :emphasize-lines: 6, 11
+
+Run or Roll Back a Migration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To run the database migration from a class file, run the following PHP artisan
+command after replacing the placeholder:
+
+.. code-block:: bash
+
+   php artisan migrate --path=<path to your migration class file>
+
+This artisan command runs the ``up()`` function in the class file to create
+the collection and index in the database specified in the ``config/database.php``
+file.
+
+To roll back the migration, run the following PHP artisan command after
+replacing the placeholder:
+
+.. code-block:: bash
+
+   php artisan migrate:rollback --path=<path to your migration class file>
+
+This command runs the ``down()`` function in the class file to drop the
+collection and related indexes.
+
+To learn more about Laravel migrations, see
+`Database: Migrations <https://laravel.com/docs/{+laravel-docs-version+}/migrations>`__
+in the Laravel documentation.
+
+.. _laravel-eloquent-collection-exists:
+
+Check Whether a Collection Exists
+---------------------------------
+
+To check whether a collection exists, call the ``hasCollection()`` method on
+the ``Schema`` facade in your migration file. You can use this to
+perform migration logic conditionally.
+
+The following example migration creates a ``stars`` collection if a collection
+named ``telescopes`` exists:
+
+.. literalinclude:: /includes/schema-builder/stars_migration.php
+   :language: php
+   :dedent:
+   :start-after: begin conditional create
+   :end-before: end conditional create
+
+.. _laravel-eloquent-indexes:
+
+Manage Indexes
+--------------
+
+MongoDB indexes are data structures that improve query efficiency by reducing
+the number of documents needed to retrieve query results. Indexes such as
+geospatial indexes extend how you can query the data.
+
+To improve query performance by using an index, make sure the index covers
+the query.  To learn more about indexes and query optimization, see the
+following MongoDB server manual entries:
```

Comment:
I think we decided on capitalizing "Server" for the Server docs (hope I'm remembering that right):
```suggestion
following MongoDB Server manual entries:
```

Reply:
I think it might be "Server manual" (omitting MongoDB, as in most of the driver docs) or "MongoDB server manual" (C# docs) from looking at the other docs repos. I'll use the former and create a source constant so that it's easy to change.